File: Algorithm/Python/789. Escape The Ghosts.py
# -*- coding: utf-8 -*-
# Created on Fri Mar 16 2018 22:23:51
# Author: WuLC
# EMail: liangchaowu5@gmail.com

# just check who can reach the target firstly
class Solution(object):
    def escapeGhosts(self, ghosts, target):
        """
        :type ghosts: List[List[int]]
        :type target: List[int]
        :rtype: bool
        """
        tar_sum = abs(target[0]) + abs(target[1])
        for g in  ghosts:
            if abs(g[0]-target[0]) + abs(g[1]-target[1]) <= tar_sum:
                return False
        return True

File: Algorithm/C++/789. Escape The Ghosts.cpp
/*
 * Created on Fri Jun 08 2018 21:22:0
 * Author: WuLC
 * EMail: liangchaowu5@gmail.com
 */

// imagine the ghost just try to reach target as fast as possible
// if the ghost reach firstly, then you will never win
class Solution 
{
    public:
        bool escapeGhosts(vector<vector<int>>& ghosts, vector<int>& target) 
        {
            int dist = std::abs(target[0]) + std::abs(target[1]);
            for(auto g:ghosts)
            {
                if (std::abs(g[0]-target[0]) + std::abs(g[1]-target[1]) <= dist) return false;
            }
            return true;
        }
};